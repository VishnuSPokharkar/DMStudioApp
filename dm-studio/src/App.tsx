import React from "react";
import logo from "./logo.svg";
import "./App.css";
import Home from "./Home/Home";
import Contact from "./Contact/Contact";
import About from "./About/About";
import Testimonials from "./Testimonials/Testimonials";
import Gallery from "./Gallery/Gallery";

const App: React.FC<{ name: string }> = (props) => {
  return (
    <div className="App">
      <Home title="Home Page"></Home>
      <Gallery title="Gallery Page"></Gallery>
      <Contact title="Contact Us"></Contact>
      <About title="About Us"></About>
      <Testimonials title="Testimonials"></Testimonials>
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" data-testid="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Hello {props.name}. Welcome to Learn React!
        </a>
      </header>
    </div>
  );
};

export default App;
