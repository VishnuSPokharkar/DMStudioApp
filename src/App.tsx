import "./App.css";
import Home from "./components/home/Home";

function App() {
  return (
    <>
      <h1>Hi Vishnu </h1>
      <Home name="Vishnu" visible={true} />
    </>
  );
}

export default App;
