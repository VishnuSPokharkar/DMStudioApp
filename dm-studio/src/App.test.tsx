import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

// Data Driven testing
const nameList = ["Dhananjay", "Vishnu", "Gauri"];

const defaultRender = (name: string) => {
  return render(<App name={name} />);
};

describe.each(nameList)("Home Page Tests", (name) => {
  it(`should render learn react link when name is: ${name}`, () => {
    defaultRender(name);
    const linkElement = screen.getByText(
      `Hello ${name}. Welcome to Learn React!`
    );
    expect(linkElement).toBeInTheDocument();
  });

  it(`should match the snapshot when name is :${name}`, () => {
    const { asFragment } = defaultRender(name);
    expect(asFragment()).toMatchSnapshot();
  });
});
