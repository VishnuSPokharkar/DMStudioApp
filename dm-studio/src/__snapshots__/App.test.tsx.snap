// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Tests should match the snapshot when name is :Dhananjay 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Hello Dhananjay. Welcome to Learn React!
      </a>
    </header>
  </div>
</DocumentFragment>
`;

exports[`Home Page Tests should match the snapshot when name is :Gauri 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Hello Gauri. Welcome to Learn React!
      </a>
    </header>
  </div>
</DocumentFragment>
`;

exports[`Home Page Tests should match the snapshot when name is :Vishnu 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Hello Vishnu. Welcome to Learn React!
      </a>
    </header>
  </div>
</DocumentFragment>
`;
